lib/shopify_app/scripttags_manager.rb

```diff
@@ -57,5 +60,11 @@ def scripttag_exists?(src)
     def current_scripttags
       @current_scripttags ||= ShopifyAPI::ScriptTag.all.index_by(&:src)
     end
+
+    def build_src(src)
+      src.respond_to?(:call) ? src.call(shop_domain) : src
+    rescue => e
+      raise CreationFailed, "cannot create script src: #{e.message}"
```

Comment:
This API is called after the app is authenticated (which is why it has to be idempotent since it will run more than once so it's not an install hook). It is triggered in the omniauth callback here: https://github.com/Shopify/shopify_app/blob/master/lib/shopify_app/sessions_concern.rb#L18